index.js

```diff
@@ -315,6 +317,35 @@ module.exports = {
     return options;
   },
 
+  _shouldIncludeDecoratorPlugins(config) {
+    let customOptions = config['ember-cli-babel'] || {};
+
+    return customOptions.disableDecoratorTransforms !== true;
+  },
+
+  _getDecoratorPlugins(config) {
+    const { hasPlugin } = require('ember-cli-babel-plugin-helpers');
+
+    // hasPlugin expects to receive a target with an options hash, which is the
+    // config. We should make it more generic upstream.
+    let target = { options: config };
+    let plugins = [];
+
+    if (
+      hasPlugin(target, '@babel/plugin-proposal-decorators')
+      || hasPlugin(target, '@babel/plugin-proposal-class-properties')
+    ) {
+      this.project.ui.writeWarnLine(`${
+        this._parentName()
+      } has added the decorators and/or class properties plugins to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms. Ember supports the stage 1 decorator spec and transforms, so if you were using stage 2, you'll need to ensure that your decorators are compatible, or convert them to stage 1.`);
```

Comment:
I am slightly worried by line-noise cause by this warning, i have some questions:

* how often does this fire for a large app (with many nested ember-cli-babel's)?
* can it be reduced further, and maybe more meaty details to a readme?
* it should likely surface what version of ember-cli-babel now provides this via
* where does a user remove the decorators from, their add-on or their app or...

Reply:
Those are good points. Do you think we should only warn if it's a top level thing? So, apps will warn, and addons will only warn if they are being worked on/developed. I've found this a good way to move the ecosystem forward in the past when things aren't incredibly pressing, and I think this qualifies since the transforms do tend to be limited to a single package.